File: Nature of Code/PJ5/Unit 0 - Randomness/Randon Walker Big Steps/walker.js
/* Utiliza una distribución de probabilidad personalizada para variar el 
tamaño de un paso dado por el aleatorio Walker. El tamaño del paso puede
determinarse influyendo en el rango de valores elegidos.*/

class Walker{
    constructor(tam){
        this.x = width / 2;
        this.y = height / 2;
        this.tam = tam;
    }
    walk(){
        let velx;
        let vely;
        let step = this.#acceptReject();
        if(step < 0.8){
            velx = random(-10, 10);
            vely = random(-10, 10);
        }
        else{
            velx = random(-1 * step, 1 * step);
            vely = random(-1 * step, 1 * step);
        }
        this.x += velx;
        this.y += vely;
    }
    show(){
        noStroke();
        stroke(0);
        fill(51);
        if(this.x > width) this.x = 0;
        if(this.y > height) this.y = 0;
        if(this.x < 0) this.x = width - this.tam;
        if(this.y < 0) this.y = height - this.tam;
        circle(this.x, this.y, this.tam);
    }
    /*/Permite escoger la probabilidad que un número salga más
     que otro en un proceso aleatorio. */
    #acceptReject(){
        let r1 = 0;
        let r2 = 1;
        while(r2 > r1 * r1 / r1){
            r1 = random(1);
            r2 = random(1);
        }
        return r1;
    }
}

File: Nature of Code/Pygame/Unit 0/Random Walker Big Steps/walker.py
""" Utiliza una distribución de probabilidad personalizada para variar el 
tamaño de un paso dado por el aleatorio Walker. El tamaño del paso puede
 determinarse influyendo en el rango de valores elegidos."""

from random import uniform, randint
import pygame

class Walker:
    """Crea un objeto (círculo) que recorre la pantalla en base movimientos
    aleatorios creados por un algoritmo de aceptación y rechazo. """
    def __init__(self, x, y, tam):
        self.__x = x
        self.__y = y
        self.__tam = tam
    def walk(self):
        """Actualiza la posición del objeto en pantalla"""
        step: float = self.__accept_reject()
        if step < 0.8:
            vel_x = randint(-10, 10)
            vel_y = randint(-10, 10)
        else:
            vel_x = randint(-1, 1)
            vel_y = randint(-1, 1)
        self.__x += vel_x
        self.__y += vel_y        
    def show(self, window):
        """Muestra el objeto en pantalla"""
        pygame.draw.circle(window, (51, 51, 51), (self.__x, self.__y), self.__tam)
    def __accept_reject(self)->float:
        """Algoritmo de aceptación y rechazo"""
        while True:
            r1: float = uniform(-1, 1)
            r2: float = uniform(-1, 1)
            if r2 < r1 * r1 / r2:
                return r1
